internal/pkg/service/cli/cmd/remote/table/download.go

```diff
@@ -0,0 +1,100 @@
+package table
+
+import (
+	"time"
+
+	"github.com/keboola/go-client/pkg/keboola"
+	"github.com/spf13/cobra"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/helpmsg"
+	common "github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/file/download"
+	"github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/table/unload"
+)
+
+func DownloadCommand(p dependencies.Provider) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   `download [table]`,
+		Short: helpmsg.Read(`remote/table/download/short`),
+		Long:  helpmsg.Read(`remote/table/download/long`),
+		Args:  cobra.MaximumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) (cmdErr error) {
+			// Ask for host and token if needed
+			baseDeps := p.BaseDependencies()
+			if err := baseDeps.Dialogs().AskHostAndToken(baseDeps); err != nil {
+				return err
+			}
+
+			// Get dependencies
+			d, err := p.DependenciesForRemoteCommand(common.WithoutMasterToken())
+			if err != nil {
+				return err
+			}
+
+			// Send cmd successful/failed event
+			defer d.EventSender().SendCmdEvent(d.CommandCtx(), time.Now(), &cmdErr, "remote-table-unload")
+
+			var tableID keboola.TableID
+			if len(args) == 0 {
+				allTables, err := d.KeboolaProjectAPI().ListTablesRequest(keboola.WithColumns()).Send(d.CommandCtx())
+				if err != nil {
+					return err
+				}
+
+				table, err := d.Dialogs().AskTable(d.Options(), *allTables)
+				if err != nil {
+					return err
+				}
+				tableID = table.ID
+			} else {
+				id, err := keboola.ParseTableID(args[0])
+				if err != nil {
+					return err
+				}
+				tableID = id
+			}
```

Comment:
Tohle mi přišlo povědomý a teď koukám že už to tam máme pětkrát. 😅 
- https://github.com/keboola/keboola-as-code/blob/e079be34fbecc0fa13bd9afa9a1d769b330706ca/internal/pkg/service/cli/cmd/remote/table/detail.go#L37-L55
- https://github.com/keboola/keboola-as-code/blob/e079be34fbecc0fa13bd9afa9a1d769b330706ca/internal/pkg/service/cli/cmd/remote/table/import.go#L40-L69
- https://github.com/keboola/keboola-as-code/blob/e079be34fbecc0fa13bd9afa9a1d769b330706ca/internal/pkg/service/cli/cmd/remote/table/unload.go#L40-L58
- https://github.com/keboola/keboola-as-code/blob/e079be34fbecc0fa13bd9afa9a1d769b330706ca/internal/pkg/service/cli/cmd/remote/table/preview.go#L43-L60
- https://github.com/keboola/keboola-as-code/blob/e079be34fbecc0fa13bd9afa9a1d769b330706ca/internal/pkg/service/cli/cmd/remote/table/upload.go#L48-L77

Úplně stejný to není, je tam pár rozdílů, ale myslíš že by dávalo smysl to abstrahovat do společnýho kódu?

Reply:
Jo, jsou tady jen dvě varianty, data -> table (`upload`/`import`) a table -> data (`preview`, `detail`, `unload`, `download`). Možná by to šlo nějak vytáhnout, ale nevím, kam bych to dal. Do dialogu asi ne, když to dělá i API request. Nebo jo?

Reply:
🤔 no to je pravda že do dialogu by to asi jít nemělo. Co myslíš @michaljurecko ?

Reply:
Co tak `remote/table/utils.go`? Nejaka nova funkcia?

Reply:
(`utils` pkg je antipatern, kazdy package ma mat signle responsibility, ale `utils.go` subor je ok, nejaky spolocny kod vramci balicka)